src/main/java/Parser.java

```diff
@@ -0,0 +1,169 @@
+public class Parser {
+    public void parse(String prompt, TaskList tasks, Ui ui, Storage storage) throws InvalidArgumentException, InvalidTaskException, InvalidSerialException {
```

Comment:
Perhaps consider splitting this line into two? The line is current 156 characters long, which exceeds the course's hard limit of 120.

More details here: https://se-education.org/guides/conventions/java/intermediate.html#layout